src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -120,6 +120,72 @@ public int RemovePackageReference(string projectPath, LibraryDependency libraryD
             }
         }
 
+        /// <summary>
+        /// Check if the project files format are correct for CPM
+        /// </summary>
+        /// <param name="packageReferenceArgs">Arguments used in the command</param>
+        /// <returns></returns>
+        public bool AreCentralVersionRequirementsSatisfied(PackageReferenceArgs packageReferenceArgs)
+        {
+            var project = GetProject(packageReferenceArgs.ProjectPath);
+            string directoryPackagesPropsPath = project.GetPropertyValue(DirectoryPackagesPropsPathPropertyName);
+
+            IEnumerable<ProjectItem> packageReferences = project.Items.Where(i => i.ItemType == PACKAGE_REFERENCE_TYPE_TAG && i.EvaluatedInclude.Equals(packageReferenceArgs.PackageId));
+            ProjectItem packageReference = packageReferences.LastOrDefault();
+
+            // Check if the package reference is defined in the project file
+            if (packageReferences != null)
+            {
+                foreach (ProjectItem reference in packageReferences)
+                {
+                    if (!reference.Xml.ContainingProject.FullPath.Equals(project.FullPath))
+                    {
+                        packageReferenceArgs.Logger.LogError(string.Format(CultureInfo.CurrentCulture, Strings.Error_AddPkg_CentralPackageVersions_PackageReference_WrongLocation, packageReferenceArgs.PackageId));
+                        return false;
+                    }
+                }
+            }
+
+            // Get VersionOverride if it exisits in the package reference.
```

Comment:
Can we use a similar logic like restore?

I don't necessarily think we should expose public API and call it from here, but given that they are trying to assert the same thing, they should have the same code.

Reply:
@nkolev92 In this case we dont want to look at the dgspec like we do in restore, I know this command uses a dgspec we dont know if it is up to date with the project file, since this tools is modifying the project XML, that should be the source of truth.

The user can modify the XML without doing a restore and try to add a package.

Reply:
The dgspec is generated prior to the add package command running, so whether a restore was run shouldn't be relevant: https://github.com/dotnet/sdk/blob/a3d79ddc395389d0fe3f3ff9023bbf492323a606/src/Cli/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs#L71-L99

Reply:
I changed some of the functionality to check in the dgspec, I'm still checking in csproj and Directory.Package.props files to see if the property comes from the desired file, I wasnt able to check this from the dgspec